### PR TITLE
Batch Size Configuration into Manager 

### PR DESCRIPTION
**Pull request by trthhrtz**

### CODE REVIEW

File: src/main/scala/io/hydrosphere/serving/manager/domain/monitoring/MonitoringConfiguration.scala [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package io.hydrosphere.serving.manager.domain.monitoring
+
+import spray.json._
+
+case class MonitoringConfiguration(batch_size: Int = 100) {
```
Comment (KineticCookie) [inline src/main/scala/io/hydrosphere/serving/manager/domain/monitoring/MonitoringConfiguration.scala:5]: remove curly braces

File: src/main/scala/io/hydrosphere/serving/manager/domain/monitoring/MonitoringConfiguration.scala [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+}
+
+
+object MCProtocol extends DefaultJsonProtocol {
```
Comment (KineticCookie) [inline src/main/scala/io/hydrosphere/serving/manager/domain/monitoring/MonitoringConfiguration.scala:10]: Weird object name.
Make object companion instead

File: src/main/scala/io/hydrosphere/serving/manager/domain/monitoring/MonitoringConfiguration.scala [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  implicit val MCFormat = jsonFormat1(MonitoringConfiguration)
+}
+
+object MCDefault {
+  val JsValue = """{ "batch_size": 100 }""".parseJson
```
Comment (KineticCookie) [inline src/main/scala/io/hydrosphere/serving/manager/domain/monitoring/MonitoringConfiguration.scala:15]: Incorrect initialization here.

We need default value for type MonitoringConfiguration. Create it explicitly, and make as `val` in object companion.

File: src/main/scala/io/hydrosphere/serving/manager/infrastructure/db/repository/DBModelVersionRepository.scala [hunk 4]
```diff
@@ -267,16 +290,18 @@ object DBModelVersionRepository {
     }
 
     override def create(entity: ModelVersion): F[ModelVersion] = {
+      print("hello")
```
Comment (KineticCookie) [inline src/main/scala/io/hydrosphere/serving/manager/infrastructure/db/repository/DBModelVersionRepository.scala:293]: remove this

File: src/main/resources/db/migration/V11__monitoring_configuration.sql [hunk 5]
```diff
@@ -0,0 +1 @@
+ALTER TABLE hydro_serving.model_version ADD COLUMN monitoring_configuration JSON NOT NULL DEFAULT '{"batch_size": 100}'::json;
```
Comment (KineticCookie) [inline src/main/resources/db/migration/V11__monitoring_configuration.sql:1]: for backward compat we need to put 10 here
this ensures that old models that are already deployed won’t break (hopefully)

File: src/main/scala/io/hydrosphere/serving/manager/domain/monitoring/MonitoringConfiguration.scala (hunk 1 above)
Comment (KineticCookie) [reply src/main/scala/io/hydrosphere/serving/manager/domain/monitoring/MonitoringConfiguration.scala:5]:
```suggestion
case class MonitoringConfiguration(batchSize: Int = 100) {
```

File: src/main/resources/db/migration/V11__monitoring_configuration.sql (hunk 5 above)
Comment (trthhrtz) [reply src/main/resources/db/migration/V11__monitoring_configuration.sql:1]: Done

File: src/main/scala/io/hydrosphere/serving/manager/infrastructure/db/repository/DBModelVersionRepository.scala (hunk 4 above)
Comment (trthhrtz) [reply src/main/scala/io/hydrosphere/serving/manager/infrastructure/db/repository/DBModelVersionRepository.scala:293]: Done

File: src/main/scala/io/hydrosphere/serving/manager/domain/monitoring/MonitoringConfiguration.scala (hunk 1 above)
Comment (trthhrtz) [reply src/main/scala/io/hydrosphere/serving/manager/domain/monitoring/MonitoringConfiguration.scala:5]: Done

File: src/main/scala/io/hydrosphere/serving/manager/domain/monitoring/MonitoringConfiguration.scala (hunk 3 above)
Comment (trthhrtz) [reply src/main/scala/io/hydrosphere/serving/manager/domain/monitoring/MonitoringConfiguration.scala:15]: Added, could you check a new version?

File: src/main/scala/io/hydrosphere/serving/manager/domain/monitoring/MonitoringConfiguration.scala (hunk 2 above)
Comment (trthhrtz) [reply src/main/scala/io/hydrosphere/serving/manager/domain/monitoring/MonitoringConfiguration.scala:10]: I was following spray tutorial https://github.com/spray/spray-json

Could you check a new version? If not, I can move it inside MonitoringConfiguration companion object.

File: src/main/scala/io/hydrosphere/serving/manager/domain/monitoring/MonitoringConfiguration.scala [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+import DefaultJsonProtocol._
+
+case class MonitoringConfiguration(batchSize: Int)
+
+object MonitoringConfiguration {
+  val defaultBatchSize: Int = 100
+
+  def apply(batchSize: Int = defaultBatchSize): MonitoringConfiguration = {
+    new MonitoringConfiguration(batchSize)
+  }
```
Comment (KineticCookie) [inline src/main/scala/io/hydrosphere/serving/manager/domain/monitoring/MonitoringConfiguration.scala:13]:
```suggestion

```

File: src/main/scala/io/hydrosphere/serving/manager/domain/monitoring/MonitoringConfiguration.scala [hunk 7]
```diff
@@ -0,0 +1,16 @@
+package io.hydrosphere.serving.manager.domain.monitoring
+
+import spray.json._
+import DefaultJsonProtocol._
+
+case class MonitoringConfiguration(batchSize: Int)
```
Comment (KineticCookie) [inline src/main/scala/io/hydrosphere/serving/manager/domain/monitoring/MonitoringConfiguration.scala:6]:
```suggestion
case class MonitoringConfiguration(batchSize: Int = 100)
```

File: src/main/scala/io/hydrosphere/serving/manager/domain/monitoring/MonitoringConfiguration.scala (hunk 6 above)
Comment (KineticCookie) [reply src/main/scala/io/hydrosphere/serving/manager/domain/monitoring/MonitoringConfiguration.scala:13]: Unnecessary apply method

File: src/main/scala/io/hydrosphere/serving/manager/infrastructure/db/repository/DBModelVersionRepository.scala [hunk 8]
```diff
@@ -37,7 +40,8 @@ object DBModelVersionRepository {
     profile_types: Option[String],
     install_command: Option[String],
     metadata: Option[String],
-    is_external: Boolean
+    is_external: Boolean,
+    monitoring_configuration: JsValue = MonitoringConfiguration().toJson
```
Comment (KineticCookie) [inline src/main/scala/io/hydrosphere/serving/manager/infrastructure/db/repository/DBModelVersionRepository.scala:44]: No need for default value here.

**Review by KineticCookie — CHANGES_REQUESTED** [review]

File: src/main/scala/io/hydrosphere/serving/manager/domain/monitoring/MonitoringConfiguration.scala (hunk 6 above)
Comment (trthhrtz) [reply src/main/scala/io/hydrosphere/serving/manager/domain/monitoring/MonitoringConfiguration.scala:13]: Fixed

File: src/main/scala/io/hydrosphere/serving/manager/domain/monitoring/MonitoringConfiguration.scala (hunk 7 above)
Comment (trthhrtz) [reply src/main/scala/io/hydrosphere/serving/manager/domain/monitoring/MonitoringConfiguration.scala:6]: Added

File: src/main/scala/io/hydrosphere/serving/manager/infrastructure/db/repository/DBModelVersionRepository.scala (hunk 8 above)
Comment (trthhrtz) [reply src/main/scala/io/hydrosphere/serving/manager/infrastructure/db/repository/DBModelVersionRepository.scala:44]: Fixed

**Comment by trthhrtz** [comment]
Conflicts resolved

File: src/main/scala/io/hydrosphere/serving/manager/domain/monitoring/MonitoringConfiguration.scala [hunk 9]
```diff
@@ -0,0 +1,12 @@
+package io.hydrosphere.serving.manager.domain.monitoring
+
+import spray.json._
+import DefaultJsonProtocol._
+
+case class MonitoringConfiguration(batchSize: Int = 100)
+
+object MonitoringConfiguration {
+  val defaultBatchSize: Int = 100
```
Comment (KineticCookie) [inline src/main/scala/io/hydrosphere/serving/manager/domain/monitoring/MonitoringConfiguration.scala:9]: Is it used anywhere?

Comment (trthhrtz) [reply src/main/scala/io/hydrosphere/serving/manager/domain/monitoring/MonitoringConfiguration.scala]: Deleted

**Comment by trthhrtz** [comment]
@KineticCookie fixed, tested with adult from private examples

File: src/main/scala/io/hydrosphere/serving/manager/domain/model/ModelService.scala [hunk 10]
```diff
@@ -182,6 +183,8 @@ object ModelService {
           modelContract = modelReq.contract,
           model = parentModel,
           metadata = modelReq.metadata.getOrElse(Map.empty),
+          // FIXME: Where to get MC?
+          monitoringConfiguration = MonitoringConfiguration()
```
Comment (trthhrtz) [inline src/main/scala/io/hydrosphere/serving/manager/domain/model/ModelService.scala:187]: @KineticCookie could you revise this please?

File: src/main/scala/io/hydrosphere/serving/manager/domain/model_build/ModelVersionBuilder.scala [hunk 11]
```diff
@@ -58,7 +59,8 @@ object ModelVersionBuilder {
           status = ModelVersionStatus.Assembling,
           installCommand = metadata.installCommand,
           metadata = metadata.metadata,
-        )
+          // FIXME: Where to get MC?
+          monitoringConfiguration = MonitoringConfiguration())
```
Comment (trthhrtz) [inline src/main/scala/io/hydrosphere/serving/manager/domain/model_build/ModelVersionBuilder.scala:63]: @KineticCookie and here

**Pull request merged**